docs_src/design/adr/0005-Service-Self-Config.md

```diff
@@ -0,0 +1,118 @@
+
+The implementation for self-seeding services and environmental overrides is already implemented (for Fuji) per this document in the application services and device services (and instituted in the SDKs of each).
+
+## Backward compatibility
+Several aspects of this ADR contain backward compatibility issues for the device service and application service SDKs.  Therefore, for the upcoming minor release, the following guidelines and expections are added to provide for backward compatibility.
+
+- --registry=<url> for Device SDK
+        
+If used with URL , assume previous version and use URL for --configProvider option and default to http as DS SDK does today. Also use URL for registry connection.
+
+- --registry (no ‘=’) and w/o --configProvider for both SDKs
+
+Assume for this release that one would never use registry w/o config provider.
+Use registry configuration from file for config provider as is done in Fuji. 
+
+- Env Var: edgex_registry=<url> for all services (currently has been removed)
+
+Add it back and use value as if it was edgex_configuration_provider and enable use of registry with same settings in URL. Default to http as it is in Fuji.
+ 
+## Consequences 
+
+- Docker compose files will need to be changed to remove config seed. 
+- The main Snap will need to be changed to remove config seed. 
+- Config seed code (currently in edgex-go repo) is to be removed.
+- Any service specific environmental overrides currently on config seed need to be moved to the specific service(s).
+- The Docker configuration files and directory (example:  https://github.com/edgexfoundry/edgex-go/blob/master/cmd/core-data/res/docker/configuration.toml) that are used to populate the config seed for Docker containers can be eliminated from all the services.
+- Documentation would need to reflect removal of config seed and "self seeding" process.
+- Removes any potential issue with past race conditions (as experienced with the Edinburgh release) as each service is now responsible for its own configuration.
+  > There are still high availability concerns that need to be considered and not covered in this ADR at this time.
+- Removes some confusion on the part of users as to why a service (config-seed) starts and immediately exits.
+- Minimal impact to development cycles and release schedule 
+- Docker files will need to be modified to remove setting profile=docker
+- Docker compose files will need to be changed to add environmental overrides for removal of docker profiles. These should go in the global environment section of the compose files for those overrides that apply to all services.  Example:
+~~~~~
+# all common shared environment variables defined here:
+x-common-env-variables: &common-variables
+  EDGEX_SECURITY_SECRET_STORE: "false"
+  edgex_configuration_provider: consul.http://edgex-core-consul:8500
```

Comment:
I'd like to see this upper-cased.

Reply:
@tonyespy , agreed since this one was added for Geneva it should be changed to upper-case. Other legacy env var like edgex_registry will remain lower case for Geneva.

Reply:
This will be upper case; other config may be lowercase to remain backward compatible.